markdown/dt-otel/dt-otel.md

```diff
@@ -779,6 +808,176 @@ The second span was generated by the PubSub+ Broker when the message was receive
+    //Create a child span and set extracted/current context as parent of this span
+    final Span sendSpan = JcsmpTracingUtil.tracer
+            .spanBuilder(SERVICE_NAME + " " + SpanAttributes.MessagingOperation.PROCESS)
+            .setSpanKind(SpanKind.CLIENT)
+            .setAttribute(SpanAttributes.MessagingAttribute.DESTINATION_KIND.toString(),
+                    SpanAttributes.MessageDestinationKind.TOPIC.toString())
+            .setAttribute(SpanAttributes.MessagingAttribute.IS_TEMP_DESTINATION.toString(), "true")
+            //Set more attributes as needed
+            .setAttribute("myKey", "myValue" + ThreadLocalRandom.current().nextInt(1, 3))
+            .setParent(extractedContext) // set extractedContext as parent
+            .startSpan();
+// continues below...
+```
+
+After creating the new span, we are able to inject it back on the message as now current context. 
+After which we go ahead and send our message, making sure to catch relevant errors and end the current span. Only after ending the current span
+is the data exported to the configured Opentelemetry collector.
+<aside>
+The OpenTelemetry collector is brought in to the application via the _opentelemetry-exporter-otlp_ library. Default settings are used to export
+to the collector executing in Docker.
+</aside>
+
+```java
+// ...continued from above
+    try (Scope scope = sendSpan.makeCurrent()) {
+        final SolaceJCSMPTextMapSetter setter = new SolaceJCSMPTextMapSetter();
+        final TextMapPropagator propagator = JcsmpTracingUtil.openTelemetry.getPropagators().getTextMapPropagator();
+        //and then inject new current context (set by sendSpan.makeCurrent()) in the message
+        propagator.inject(Context.current(), message, setter);
+        try {
+            producer.send(message, topic);  // send the message
+            msgSentCounter++;  // add one
+            message.reset();   // reuse this message, to avoid having to recreate it: better performance
+        } catch (JCSMPException e) {
+            System.out.printf("### Caught while trying to producer.send(): %s%n", e);
+            if (e instanceof JCSMPTransportException) {  // all reconnect attempts failed
+            isShutdown = true;  // let's quit; or, could initiate a new connection attempt
+        }
+            throw e;
+        }
+    } catch (Exception e) {
+        sendSpan.recordException(e); //Span can record exception if any
+        sendSpan.setStatus(StatusCode.ERROR, e.getMessage()); //Set span status as ERROR/FAILED
+    } finally {
+        sendSpan.end(); //End sendSpan. Span data is exported when span.end() is called.
+    }
+}
+```
+
+We've provided a jar containing the manual implementation code from above, to run the jar and publish messages to the broker, execute the command below from the tracing-codelab directory.
+```bash
+java -jar solace-samples-jcsmp-publisher-manual-instrumentation-1.0-SNAPSHOT.jar tcp://localhost:55557 default default default
```

Comment:
File is currently available [here](https://solacesystems-my.sharepoint.com/personal/hank_spencer_solace_com/_layouts/15/onedrive.aspx?FolderCTID=0x0120000F184F3829FEEA4782B84E73C4226B07&id=%2Fpersonal%2Fhank%5Fspencer%5Fsolace%5Fcom%2FDocuments%2FDT%20OTEL%20Code%20Lab%2Fmanual%2Dinstrumentation%2Dpublisher%2Ftarget&view=0). We'll need to copy the final version of both the unpackaged code and the jar to the tracing-codelab zip that is currently downloaded at the start of the codelab.

Unpackaged code can be ran as a java class. I'm assuming we'll want to provide the users with a .jar file so they don't have to package anything up.

Reply:
We will also need readme instructions on how to compile the java code (maven or gradle)